llvm/lib/Analysis/LazyValueInfo.cpp

```diff
@@ -1932,6 +1932,32 @@ LazyValueInfo::Tristate LazyValueInfo::getPredicateAt(unsigned P, Value *LHS,
   return LazyValueInfo::Unknown;
 }
 
+LazyValueInfo::Tristate
+LazyValueInfo::getPredicateAtUse(unsigned P, const Use &LHS, const Use &RHS) {
+  const Instruction *CxtI = dyn_cast<Instruction>(LHS.getUser());
+
+  if (!CxtI)
+    return LazyValueInfo::Unknown;
```

Comment:
```suggestion
  const Instruction *CxtI = cast<Instruction>(LHS.getUser());
```
The cast should never fail.